gammapy/makers/background/phase.py

```diff
@@ -100,20 +112,37 @@ def run(self, dataset, observation):
         counts_off = self.make_counts_off(dataset, observation)
         counts = self.make_counts(dataset, observation)
 
-        acceptance = RegionNDMap.from_geom(geom=dataset.counts.geom)
-        acceptance.data = np.sum([_[1] - _[0] for _ in self.on_phase])
+        if isinstance(dataset, SpectrumDataset):
+            acceptance = RegionNDMap.from_geom(geom=dataset.counts.geom)
+            acceptance.data = np.sum([_[1] - _[0] for _ in self.on_phase])
 
-        acceptance_off = RegionNDMap.from_geom(geom=dataset.counts.geom)
-        acceptance_off.data = np.sum([_[1] - _[0] for _ in self.off_phase])
+            acceptance_off = RegionNDMap.from_geom(geom=dataset.counts.geom)
+            acceptance_off.data = np.sum([_[1] - _[0] for _ in self.off_phase])
 
-        dataset_on_off = SpectrumDatasetOnOff.from_spectrum_dataset(
-            dataset=dataset,
-            counts_off=counts_off,
-            acceptance=acceptance,
-            acceptance_off=acceptance_off,
-        )
-        dataset_on_off.counts = counts
-        return dataset_on_off
+            dataset_on_off = SpectrumDatasetOnOff.from_spectrum_dataset(
```

Comment:
Maybe simplify a bit by
```
dataset_on_off = MapDatasetOnOff.from_spectrum_dataset(
             dataset=dataset,
             counts_off=counts_off,
             acceptance=acceptance,
             acceptance_off=acceptance_off,
         )

if `dataset._geom.is_region:
     dataset_on_off = dataset_on_off.to_spectrum_dataset(dataset._geom.region)

return dataset_on_off
```

Reply:
Hi @AtreyeeS,
I have made all the modifications that you suggest apart from this one. 
If I understand correctly what you suggest, it means that in both case (SpectrumonOff or MapOnOff), the user has to pass a MapDataset to the maker? (I guessed that it MapDatasetOnOff.from_map_dataset([...]) and not MapDatasetOnOff.from_spectrum_dataset([...])).
If I understand correctly, I'm not sure that it is very intuitive for the user to build a MapDataset and then to have the maker transform it into a SpectrumDataset if the user wants to do a SpectrumDataset from the start. 
Let me know

Reply:
Hi @MRegeard ! Yes indeed, it should be `MapDatasetOnOff.from_map_dataset` as you correctly pointed out. Sorry! 

The user does not need to create a `MapDataset` explicitly.
`MapDatasetOfOff.xyz()` functionalities are available on `SpectrumDataset` due to inheritance. So, `MapDatasetOnOff.from_map_dataset` will create a MapDatasetOnOff. Now, if dataset was initially a spectral one to start off with, gammapy will change to a spectral datasetonoff

Reply:
Thinking a bit about this, I think it is better to change 

```
if dataset._geom.is_region:
     dataset_on_off = dataset_on_off.to_spectrum_dataset(dataset._geom.region)
```

to 

```
if isInstance(dataset, SpectrumDataset):
    dataset_on_off = dataset_on_off.to_spectrum_dataset(dataset._geom.region)
```

Reply:
Oh I see, I always forget about inheritance...
Changes were made !